include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -135,6 +135,22 @@ inline dim<2> get_size(const dim<2> &size) { return size; }
     }
 
 
+/**
+ *Asserts that _op1 is a non-empty matrix.
+ *
+ *@throw BadDimension if the number of rows of _op1 and the number of
+ *                          columns of _op1 is equal to zero.
+ */
+#define GKO_ASSERT_IS_NON_EMPTY_MATRIX(_op1)                           \
+    if (::gko::detail::get_size(_op1)[0] == 0 &&                       \
+        ::gko::detail::get_size(_op1)[1] == 0) {                       \
```

Comment:
I would highly prefer if you would use the `bool` cast option from `gko::dim` (see implementation [here](https://github.com/ginkgo-project/ginkgo/blob/a686faeb4c6276c5b613e3a01c269bb62545436e/include/ginkgo/core/base/dim.hpp#L120)):
```
if (::gko::detail::get_size(_op1))
```
That makes sure that as soon as one dimension is zero, the matrix is considered empty (which it would then be).

If you want to stick with the current version, replace `&&` with `||` since a matrix is empty as soon as one dimension is `0`.